pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -73,117 +85,156 @@ def __init__(self):
             print("\n")
         return 1
 
-    def create_instr_list(self, cfg):
-        self.instr_names.clear()
-        self.instr_group.clear()
-        self.instr_category.clear()
-        for instr_name, values in self.instr_registry.items():
+    @classmethod
+    def create_instr_list(cls, cfg):
+        cls.instr_names.clear()
+        cls.instr_group.clear()
+        cls.instr_category.clear()
+        for instr_name, values in cls.instr_registry.items():
             if instr_name in rcs.unsupported_instr:
                 continue
-            instr_inst = self.create_instr(instr_name)
-            self.instr_template[instr_name] = instr_inst
+            instr_inst = cls.create_instr(instr_name)
+            cls.instr_template[instr_name] = instr_inst
 
             if not instr_inst.is_supported(cfg):
                 continue
-            if ((rcs.XLEN != 32) and (instr_name == "C_JAL")):
+            if ((rcs.XLEN != 32) and (instr_name == riscv_instr_name_t.C_JAL)):
                 continue
-            if ((riscv_reg_t.SP in cfg.reserved_regs) and (instr_name == "C_ADDI16SP")):
+            if ((riscv_reg_t.SP in cfg.reserved_regs) and
+                    (instr_name == riscv_instr_name_t.C_ADDI16SP)):
                 continue
-            if (cfg.enable_sfence and instr_name == "SFENCE_VMA"):
+            if (cfg.enable_sfence and instr_name == riscv_instr_name_t.SFENCE_VMA):
                 continue
-            if instr_name in ["FENCE", "FENCE_I", "SFENCE_VMA"]:
+            if instr_name in [riscv_instr_name_t.FENCE, riscv_instr_name_t.FENCE_I,
+                              riscv_instr_name_t.SFENCE_VMA]:
                 continue
             if (instr_inst.group.name in rcs.supported_isa and
-                    not(cfg.disable_compressed_instr and
-                        instr_inst.group in ["RV32C", "RV64C", "RV32DC", "RV32FC", "RV128C"]) and
-                    not(not(cfg.enable_floating_point) and instr_inst.group in
-                        ["RV32F", "RV64F", "RV32D", "RV64D"])):
-                self.instr_category[instr_inst.category.name].append(instr_name)
-                self.instr_group[instr_inst.group.name].append(instr_name)
-                self.instr_names.append(instr_name)
-        self.build_basic_instruction_list(cfg)
-        self.create_csr_filter(cfg)
-
-    def create_instr(self, instr_name):
-        """TODO This method is specific to RV32I instruction only.
-        It must be scaled to all instruction extensions."""
+               not(cfg.disable_compressed_instr and instr_inst.group in [riscv_instr_group_t.RV32C,
+               riscv_instr_group_t.RV64C, riscv_instr_group_t.RV32DC, riscv_instr_group_t.RV32FC,
+               riscv_instr_group_t.RV128C] and not(not(cfg.enable_floating_point) and
+               instr_inst.group in [riscv_instr_group_t.RV32F, riscv_instr_group_t.RV64F,
+               riscv_instr_group_t.RV32D, riscv_instr_group_t.RV64D]))):
```

Comment:
This condition looks weird (too long in terms of appearance), would be great if you can cut the enum name (riscv_instr_group_t), or something similar.